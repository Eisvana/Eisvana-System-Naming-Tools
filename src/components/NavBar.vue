<script setup lang="ts">
import ThemeSwitch from './ThemeSwitch.vue';

const isEisvanaHost = window.location.host === 'naming.eisvana.com';
</script>

<template>
  <nav>
    <ul>
      <li>
        <a :href="isEisvanaHost ? 'https://eisvana.com' : '..'">&larr; View other pages</a>
      </li>
    </ul>
    <ul>
      <li>
        <ThemeSwitch />
      </li>
    </ul>
  </nav>
</template>
