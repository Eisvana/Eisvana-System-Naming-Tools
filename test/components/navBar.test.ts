import NavBar from '@/components/NavBar.vue';
import { beforeEach, describe, expect, it } from 'vitest';
import { mount, type VueWrapper } from '@vue/test-utils';

describe('NavBar', () => {
  let wrapper: VueWrapper;

  beforeEach(() => {
    wrapper = mount(NavBar);
  });

  it('should render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
