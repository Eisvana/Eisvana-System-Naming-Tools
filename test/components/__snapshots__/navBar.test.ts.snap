// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavBar > should render correctly 1`] = `
"<nav>
  <ul>
    <li><a href="..">← View other pages</a></li>
  </ul>
  <ul>
    <li><button data-v-7e4fbe6f="" class="themeswitcher" id="themeSwitch"> Switch Theme </button></li>
  </ul>
</nav>"
`;
